pages/transactionUpload.tsx

```diff
@@ -0,0 +1,225 @@
+        } else {
+          customMessage = (
+            <>
+              <Text weight={500}>Problem connecting to server:&nbsp;</Text>
+              <Text color="red">{error.message}</Text>
+            </>
+          );
+        }
+        cleanNotifications();
+        showNotification({ ...notifProps, message: customMessage });
+      });
+  }
+
+  //parses the response array from a transaction bundle upload and returns each response as an appropriate JSX Element
+  function processBundleResponse(responseArray: fhirJson.BundleEntry[] | null) {
+    return sortBundleArray(responseArray)?.map((el, index) => {
+      if (el?.response?.location) {
+        const FHIR_VERSION_LENGTH = 6;
+        const resourceLocation = el.response.location.substring(FHIR_VERSION_LENGTH);
+        return (
+          <div key={`${index}-${el.response.location}`}>
+            {el.response.status}:&nbsp;
+            <Link href={`/${resourceLocation}`} passHref>
+              <Text component="a" color="cyan">
+                {resourceLocation}
+              </Text>
+            </Link>
+          </div>
+        );
+      } else if (el?.response?.outcome?.resourceType === "OperationOutcome") {
+        const responseAsAny = el?.response?.outcome as fhirJson.OperationOutcome;
+        if (responseAsAny.issue != null) {
+          const issueArray = responseAsAny?.issue[0] as fhirJson.OperationOutcomeIssue;
+          return (
+            <div key={`response-${index}`}>
+              {el?.response?.status}:&nbsp;
+              <Text component="a" color="red">
+                {issueArray.details?.text}
+              </Text>
+            </div>
+          );
+        } else {
+          return el?.response?.status ? (
+            <Text key={`response-${index}`}>{el?.response?.status}</Text>
+          ) : (
+            <Text key={`response-${index}`}>Unexpected response</Text>
+          );
+        }
+      } else {
+        return <Text key={`response-${index}`}>Unexpected response</Text>;
+      }
+    });
+  }
+};
+
+/**
+ * Sorts an array of responses from a transaction bundle upload, by status code (descending).
+ * For responses of the same status that include locations, the responses are sorted alphabetically based on location
+ * @param toSort is the fhirJson.BundleEntry array that is to be sorted
+ * @returns the sorted fhirJson.BundleEntry array
+ */
+function sortBundleArray(toSort: fhirJson.BundleEntry[] | null) {
+  return toSort?.sort((a, b) => {
+    if (a?.response?.status && b?.response?.status) {
+      if (b.response.status === a.response.status) {
+        if (b?.response?.location && a?.response?.location) {
+          return compareTwoStrings(a.response.location, b.response.location);
+        } else {
+          return 1;
+        }
+      } else {
+        return compareTwoStrings(b.response.status, a.response.status);
+      }
+    } else {
+      return 1;
+    }
+  });
```

Comment:
I think the reason I had compareTwoStrings twice was because the status codes should be in descending order but alphabetical sorting is ascending. So if I implement your recommendation, sorting by location will be in reverse alphabetical order. I'll try and see if there is still another way I can write the sortBundleArray function

Reply:
Oh god... didn't catch that!! You may have a perfectly good solution thent